cpp/src/parquet/arrow/record_reader.cc

```diff
@@ -717,11 +727,96 @@ inline void TypedRecordReader<DType>::ConfigureDictionary(const DictionaryPage*
   DCHECK(current_decoder_);
 }
 
+// If the data page includes repetition and definition levels, we
+// initialize the level decoders and return the number of encoded level bytes.
+// The return value helps determine the number of bytes in the encoded data.
+template <typename DType>
+template <typename PageType>
```

Comment:
Pretty minor nit, I prefer:

```
template <typename DType, typename PageType> 
```
over the multiple line version.

Reply:
Hi Hatem, are you suggesting putting that code on a single line, like:

`template <typename DType> template <typename PageType>`

or instead merging it into:

`template <typename DType, typename PageType>`

If I understand this correctly, I don't think the latter will work here since this is a [member function template definition in a class template](https://stackoverflow.com/questions/5108359/how-do-i-define-a-template-function-within-a-template-class-outside-of-the-class). 

To make an analogy to the value-level, the latter option would be like defining a function that requires an `f(a)(b)` signature using `f(a, b)` instead.

Reply:
Yep, my bad.  Ignore my wrong advice.